### PR TITLE
feat: replace donate-form with multi-tier-donation-form 

### PR DESCRIPTION
**Pull request by ahmaxed**
- feat: create multi-tier-donation-form
- feat: replace DonateForm with MultiTierDonationForm
- feat: display maybe later button correctly.
- update third party e2e tests

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/components/Donation/donation-modal.tsx [hunk 1]
```diff
@@ -77,359 +62,69 @@ const Illustration = ({
...
 }: DonateModalProps): JSX.Element {
-  const [closeLabel, setCloseLabel] = useState(false);
-  const [ctaNumber, setCtaNumber] = useState(0);
-  const [isDisabled, setIsDisabled] = useState(true);
-  const [showSkipButton, setShowSkipButton] = useState(false);
-  const [showDonateForm, setShowDonateForm] = useState(true);
-  const [donationAmount, setDonationAmount] = useState(
-    defaultDonation.donationAmount
-  );
-  const loadElementsIndividually = useFeature('load_elements_individually').on;
```
Comment (ahmaxed) [inline client/src/components/Donation/donation-modal.tsx:410]: Removed all current AB tests in the modal

File: client/src/client-only-routes/show-certification.tsx [hunk 2]
```diff
@@ -41,6 +39,7 @@ import {
   certTypes,
   certTypeTitleMap
 } from '../../../shared/config/certification-settings';
+import MultiTierDonationForm from '../components/Donation/multi-tier-donation-form';
```
Comment (ahmaxed) [inline client/src/client-only-routes/show-certification.tsx:42]: DonateForm is replaced with MultiTierDonationForm. The multi-tier form will wrap the donate form only after tier selection

File: client/src/components/Donation/donate-form.tsx [hunk 3]
```diff
@@ -296,15 +290,8 @@ class DonateForm extends Component<DonateFormProps, DonateFormComponentState> {
   }
 
   renderPageForm() {
-    const { donationAmount } = this.state;
-    const { t } = this.props;
-    const usd = formattedAmountLabel(donationAmount);
-    const hours = convertToTimeContributed(donationAmount);
-    const donationDescription = t('donate.your-donation-2', { usd, hours });
```
Comment (ahmaxed) [inline client/src/components/Donation/donate-form.tsx:303]: Descriptions are moved to the multi-tier selection component.

File: client/src/components/Donation/donation-modal.tsx [hunk 4]
```diff
@@ -77,359 +62,69 @@ const Illustration = ({
   );
 };
 
-function getctaNumberBetween1To10() {
-  const min = 1;
-  const max = 10;
-  return Math.floor(Math.random() * (max - min + 1)) + min;
-}
```
Comment (ahmaxed) [inline client/src/components/Donation/donation-modal.tsx:84]: The new CTA is performing better to the CTA rotation feature, so all of the related logic is removed.

File: client/src/components/Donation/donation-modal.tsx [hunk 5]
```diff
@@ -77,359 +62,69 @@ const Illustration = ({
...
-    </div>
-  );
-}
-
-function SelectionTabs({
-  loadElementsIndividually,
-  donationAmount,
-  setDonationAmount,
-  setShowDonateForm
-}: {
```
Comment (ahmaxed) [inline client/src/components/Donation/donation-modal.tsx:130]: Tabs are also moved to the multi-tier.

**Review by ShaunSHamilton — APPROVED** [review]
This looks good to me 👍 

---

The only thing to mention is the tabbing behaviour seems _"off"_. Specifically, when tabbing through the donation amounts (also called 'tabs'), it appears as though they are set to "active", even though it still requires `enter` to be pressed to ACTUALLY select the amount.

_Probably something for another PR, and I am not even sure if this is related to this code_

**Comment by ahmaxed** [comment]
Thanks for catching that. It should be fixed on this pr.

**Comment by ahmaxed** [comment]
I should be resolved now.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 

---

Another a11y concern: There does not seem to be a way to focus the `edit amount` button that appears after clicking `Donate`.

**Comment by ahmaxed** [comment]
It was selecting it, but there were some conflicting styles. This should do it.

**Review by raisedadead — APPROVED** [review]

**Comment by ShaunSHamilton** [comment]
Mind resolving your conversations @ahmaxed ?

**Pull request merged**